session/manager_http.go

```diff
@@ -202,7 +202,15 @@ func (s *ManagerHTTP) FetchFromRequest(ctx context.Context, r *http.Request) (_
 		return nil, errors.WithStack(NewErrNoCredentialsForSession())
 	}
 
-	se, err := s.r.SessionPersister().GetSessionByToken(ctx, token, ExpandEverything, identity.ExpandDefault)
+	expand := identity.ExpandDefault
+	if s.r.Config().SessionWhoAmIAAL(r.Context()) == config.HighestAvailableAAL {
```

Comment:
This is a copy of the logic in `DoesSessionSatisfy`. By checking this here, we don't have to use `.Load("Credentials")` in `DoesSessionSatisfy` but can instead load it in parallel (faster) here.